### PR TITLE
fix(sql): make offset and timezone optional in sample-by queries

### LINKED ISSUE #2390: ALIGN TO CALENDAR feels buggy
### Describe the bug

`ALIGN TO CALENDAR` has a weird behaviour affecting to clauses like `FILL` or `LIMIT`. Maybe to some others, but I am guessing the fix is the same

### To reproduce

Two problems here: `LIMIT` and `FILL`

## LIMIT 
If I go to the demo site and execute this query, it works
```
select  last(price), `symbol` from trades
SAMPLE BY 1h limit 100
```

If I now add any modifiers to `SAMPLE BY,` it breaks
```
select  last(price), `symbol` from trades
SAMPLE BY 1h ALIGN to calendar limit 100
```

However, if I add a `TIME ZONE` and an `OFFSET`, it works. Note that if I add *only* a `TIME ZONE` is not enough and requires and offset even if it is zero. It seems like a parsing issue. This works

```
select  last(price), `symbol` from trades
SAMPLE BY 1h  ALIGN TO CALENDAR TIME ZONE 'Europe/Berlin'
WITH OFFSET '00:00' limit 10
```

I am aware I can use subqueries to do the limit on the outer query, but developer experience here is not great as I would expect `SAMPLE BY` and `LIMIT` either to work together or not, but not something in between

## FILL

This query works
```
select  last(price), `symbol` from trades
SAMPLE BY 1h FILL(PREV) ALIGN TO CALENDAR 
```

This query doesn't
```
select  last(price), `symbol` from trades
SAMPLE BY 1h  ALIGN TO CALENDAR  FILL(PREV)
```

No matter if I add time zone and offset. It still fails. It says `FILL` token is unexpected. Unless I am reading [the diagram at the docs wrong](https://questdb.io/docs/reference/sql/sample-by/), `FILL` can be used after a `SAMPLE BY` query

### Expected Behavior

`LIMIT` should work with `SAMPLE BY` or not, but consistently

FILL should be accepted after `ALIGN` clause according to the docs. Maybe this is just a docs issue

`TIME ZONE `and `WITH OFFSET` should be optional according to the docs, and according to the diagram at the website it seems `WITH OFFSET `shouldn't be accepted after `TIME ZONE`, but that's probably a fix for the documentation itself

### Environment

```markdown
- **QuestDB version**: 6.4.3
- **OS**: Linux
- **Browser**: Chrome
```

### Additional context

This is the architecture diagram [at the doc site](https://questdb.io/docs/reference/sql/sample-by/)

image

### PR DESCRIPTION
**Pull request by jerrinot**
fixes #2390

According to the grammar at https://questdb.io/docs/reference/sql/sample-by/#syntax when you align “to calendar” then either “time zone” or “with offset” is mandatory. 
image

So one could say the old behaviour is “working as designed” 🙈

I'd argue this behaviour is surprising. Even our own documentation uses samples where timezone/offset is missing - it works when "align to calendar" are the last tokens in a query. 
image

This change is making the timezone/offset optional in all cases, and when it's omitted then it defaults to `with offset '00:00'`.

We should also update the grammar in docs.

### CODE REVIEW

**Comment by ideoma** [comment]
Does it fix https://github.com/questdb/questdb/issues/2727 ?

**Comment by jerrinot** [comment]
It does not fix https://github.com/questdb/questdb/issues/2727 It's likely not a parsing issue.

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 11 / 11 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|11|11|100.00%|

File: core/src/main/java/io/questdb/griffin/SqlParser.java [hunk 1]
```diff
@@ -1106,23 +1106,20 @@ private void parseFromClause(GenericLexer lexer, QueryModel model, QueryModel ma
...
-
-                            if (tok != null && isWithKeyword(tok)) {
-                                tok = parseWithOffset(lexer, model);
-                            } else {
-                                model.setSampleByOffset(nextConstant("'00:00'"));
-                            }
-                        } else if (isWithKeyword(tok)) {
+                    if (tok == null) {
+                        model.setSampleByTimezoneName(null);
+                        model.setSampleByOffset(nextConstant("'00:00'"));
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/griffin/SqlParser.java:1111]: this constant can be static

**Review by bluestreak01 — CHANGES_REQUESTED** [review]

File: core/src/main/java/io/questdb/griffin/SqlParser.java (hunk 1 above)
Comment (jerrinot) [reply core/src/main/java/io/questdb/griffin/SqlParser.java:1111]: actually, it appears I can just pass `null`.

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 13 / 13 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|13|13|100.00%|

**Review by bluestreak01 — CHANGES_REQUESTED** [review]
this is probably a "fix" for something ?

**Review by bluestreak01 — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 13 / 13 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|13|13|100.00%|

**Comment by jerrinot** [comment]
flaky test re-opened: https://github.com/questdb/questdb/issues/1815

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 13 / 13 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|13|13|100.00%|

**Pull request merged**